easybuild/easyblocks/r/root.py

```diff
@@ -28,43 +28,106 @@
 @author: Kenneth Hoste (Ghent University)
 @author: Jens Timmerman (Ghent University)
 """
+from distutils.version import LooseVersion
+import os
 
-from easybuild.framework.easyconfig import MANDATORY
-from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.tools.modules import get_software_root, get_software_version
 from easybuild.tools.run import run_cmd
 
-class EB_ROOT(ConfigureMake):
+class EB_ROOT(CMakeMake):
 
     @staticmethod
     def extra_options():
         """
         Define extra options needed by Geant4
         """
         extra_vars = {
-            'arch': [None, "Target architecture", MANDATORY],
+            'arch': [None, "Target architecture", CUSTOM],
         }
-        return ConfigureMake.extra_options(extra_vars)
+        return CMakeMake.extra_options(extra_vars)
 
     def configure_step(self):
         """Custom configuration for ROOT, add configure options."""
 
-        self.cfg.update('configopts', "--etcdir=%s/etc/root " % self.installdir)
+        # using ./configure is deprecated/broken in recent versions, need to use CMake instead
+        if LooseVersion(self.version) >= LooseVersion('6.10'):
+            if self.cfg['arch']:
+                raise EasyBuildError("Specified value '%s' for 'arch' is not used, should not be set", self.cfg['arch'])
 
-        cmd = "%s ./configure %s --prefix=%s %s" % (self.cfg['preconfigopts'],
-                                                    self.cfg['arch'],
-                                                    self.installdir,
-                                                    self.cfg['configopts'])
+            cfitsio_root = get_software_root('CFITSIO')
+            if cfitsio_root:
+                self.cfg.update('configopts', '-DCFITSIO=%s' % cfitsio_root)
 
-        run_cmd(cmd, log_all=True, log_ok=True, simple=True)
+            fftw_root = get_software_root('FFTW')
+            if fftw_root:
+                self.cfg.update('configopts', '-Dbuiltin_fftw3=OFF -DFFTW_DIR=%s' % fftw_root)
+
+            gsl_root = get_software_root('GSL')
+            if gsl_root:
+                self.cfg.update('configopts', '-DGSL_DIR=%s' % gsl_root)
+
+            mesa_root = get_software_root('Mesa')
+            if mesa_root:
+                self.cfg.update('configopts', '-DDOPENGL_INCLUDE_DIR=%s' % os.path.join(mesa_root, 'include'))
+                self.cfg.update('configopts', '-DDOPENGL_INCLUDE_DIR=%s' % os.path.join(mesa_root, 'lib', 'libGL.so'))
```

Comment:
LIB?

Reply:
Should be `-DOPENGL_gl_LIBRARY`... Apparently ROOT is able to find the library regardless, but this is clearly wrong, I'll fix it.